File: 1456.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = {"a", "e", "i", "o", "u"}
        ans = 0
        curr = 0
        for i in range(k):
            if s[i] in vowels:
                curr += 1
        ans = max(ans, curr)
        for i in range(k, len(s)):
            if s[i - k] in vowels:
                curr -= 1
            if s[i] in vowels:
                curr += 1
            ans = max(ans, curr)
        return ans


print(Solution().maxVowels("tryhard", 4))

File: 1456.cpp
#include <set>
#include <string>
using namespace std;
class Solution {
private:
  bool in(char ch, set<char> &vowels) {
    return vowels.find(ch) != vowels.end();
  }

public:
  int maxVowels(string s, int k) {
    set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
    int ans = 0, curr = 0;
    for (int i = 0; i < k; ++i) {
      if (in(s[i], vowels)) {
        curr += 1;
      }
    }
    ans = max(ans, curr);
    for (int i = k; i < s.size(); ++i) {
      if (in(s[i - k], vowels)) {
        curr -= 1;
      }
      if (in(s[i], vowels)) {
        curr += 1;
      }
      ans = max(ans, curr);
    }
    return ans;
  }
};